perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -552,17 +553,14 @@ public UserExtSource getUserExtSourceByExtLogin(PerunSession sess, ExtSource sou
 	@Override
 	public List<UserExtSource> getActiveUserExtSources(PerunSession sess, User user) throws InternalErrorException {
 		//get now date
-		Calendar date = Calendar.getInstance();
-		date.add(Calendar.MONTH, -MAX_OLD_OF_ACTIVE_USER_EXTSOURCE);
-
-		// create sql toDate()
-		String compareDate = BeansUtils.getDateFormatterWithoutTime().format(date.getTime());
+		LocalDate date = LocalDate.now();
+		date = date.minusMonths(MAX_OLD_OF_ACTIVE_USER_EXTSOURCE);
 
 		try {
 			String query = "select " + userExtSourceMappingSelectQuery + ", " + ExtSourcesManagerImpl.extSourceMappingSelectQuery +
 					" from user_ext_sources left join ext_sources on user_ext_sources.ext_sources_id=ext_sources.id where " +
 					" user_ext_sources.user_id=? and " +
-					" user_ext_sources.last_access > " + Compatibility.toDate("'" + compareDate + "'", "'YYYY-MM-DD'");
+					" user_ext_sources.last_access > " + Compatibility.toDate("'" + date.toString() + "'", "'YYYY-MM-DD'");
```

Comment:
I think you do not need to explicitly call the `toString` method.

Reply:
Also, the previous version produced String containing time only. Are you sure this cast to String will produce the same result?

Reply:
Yes, that `toString` was redundant, it is fixed now. Concerning the `compareDate`, I am pretty sure it is the same String as `date.toString()` returns, that is just date, no time.